thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -44,13 +57,19 @@
   private final RawAnomalyResultManager rawAnomalyResultDAO;
   private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
   private static final ThirdEyeCacheRegistry CACHE_REGISTRY_INSTANCE = ThirdEyeCacheRegistry.getInstance();
+  private final AlertFilterAutotuneFactory alertFilterAutotuneFactory;
+  private final AlertFilterFactory alertFilterFactory;
+
+  private static final Logger LOG = LoggerFactory.getLogger(DetectionJobResource.class);
 
-  public DetectionJobResource(DetectionJobScheduler detectionJobScheduler) {
+  public DetectionJobResource(DetectionJobScheduler detectionJobScheduler, String alertFilterAutotuneConfigPath, String alertFilterConfigPath) {
```

Comment:
I would suggest that we have a AutotuneJobResource, that can be used through UI (ad-hoc) or background job scheduler (in periodic fashion even in this scenario, we are talking about weekly tune, not hourly).